apps/webapp/web/static/js/docs/index.js

```diff
@@ -0,0 +1,71 @@
+// file: docs/index.js
+import React from 'react';
+import mdlUpgrade from 'lib/mdlUpgrade';
+import material from 'material-design-lite/material.css';
+import styles from 'docs/styles.css';
+import TableOfContents from 'docs/TableOfContents';
+import DocPage from 'docs/DocPage';
+import fetch from 'isomorphic-fetch';
+
+class Index extends React.Component {
+
+  constructor(props) {
+    super();
+    this.state = {file: '', docs: []};
+    this.changeDocument = this.changeDocument.bind(this);
+  }
+
+  componentDidMount() {
+    // TODO: change window.location.origin to a URL variable rendered from the server.
+    fetch(window.location.origin + '/docs/index')
+      .then(response => {
+        if (response.status >= 400) {
+          throw new Error("Bad response from server");
+        }
+        return response.json();
+      })
+      .then(list => {
+        this.setState({ docs: list.data });
```

Comment:
@rsjavier14 If you want to DRY this too, you could have a function `fetchData(url, data)` where you can do all the fetch and use `data` as the state key: 
```
var newState = {}
newState[data] = response.data
this.setState(newState)
```

I just don't know if that is ok or if it would obfuscate the code.
What do you think @hisapy ?